src/main/java/URLLinks.java

```diff
@@ -0,0 +1,41 @@
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Logger;
+
+
+
+public class URLLinks implements Links{
+
+    protected Links <String> links;
+    Logger log = Logger.getLogger(Loggin.class.getName());
+
+    public URLLinks(Links<String> links) {
+        this.links = links;
+    }
+
+    private List<URL> prepareLinks() {
```

Comment:
@PavelNejinskiy same here http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html